sysid/scripts/time_plots.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+import json
+import pathlib
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import sys
+
+# Load data
+filename = pathlib.Path(sys.argv[1])
+
+UNIT_TO_ABBREVIATION = {
+    "Meters": "m",
+    "Feet": "ft",
+    "Inches": "in",
+    "Degrees": "deg",
+    "Rotations": "rot",
+    "Radians": "rad",
+}
+
+# Make DataFrame to facilitate plotting
+if filename.suffix == ".json":
+    raw_data = json.loads(filename.read_text())
+    unit = raw_data["units"]
+
+    # Get Unit
+    try:
+        abbreviation = UNIT_TO_ABBREVIATION[unit]
+    except KeyError:
+        raise ValueError("Invalid Unit")
+
+    # Make Columns
+    columns = ["Timestamp (s)", "Test"]
+    if "Drive" in raw_data["test"]:
+        columns.extend(
+            [
+                "Left Volts (V)",
+                "Right Volts (V)",
+                f"Left Position ({abbreviation})",
+                f"Right Position ({abbreviation})",
+                f"Left Velocity ({abbreviation}/s)",
+                f"Right Velocity ({abbreviation}/s)",
+                "Gyro Position (deg)",
+                "Gyro Rate (deg/s)",
+            ]
+        )
+        unit_columns = columns[4:8]
+    else:
+        columns.extend(
+            ["Volts (V)", f"Position ({abbreviation})", f"Velocity ({abbreviation}/s)"]
+        )
+        unit_columns = columns[3:]
+
+    prepared_data = pd.DataFrame(columns=columns)
+    for test in (test for test in raw_data.keys() if "-" in test):
+        formatted_entry = [[pt[0]] + [test] + pt[1:] for pt in raw_data[test]]
```

Comment:
Nit: `[pt[0], test, *pt[1:]]` is more readable and about 25% faster on my machine.